core/src/main/java/org/jruby/util/io/ChannelFD.java

```diff
@@ -131,7 +130,7 @@ private void initChannelTypes() {
         else chRead = null;
         if (ch instanceof WritableByteChannel) chWrite = (WritableByteChannel)ch;
         else chWrite = null;
-        if (ch instanceof SeekableByteChannel) chSeek = (SeekableByteChannel)ch;
```

Comment:
The reason for this is that SeekableByteChannel was added to Android 7.0 API level 24 ref.:

https://developer.android.com/reference/java/nio/channels/SeekableByteChannel.html

FileChannel has been in Android since the initial release.

I agree that using the interface is the better option, and we can do a patch in Ruboto if necessary, although better if we don't have to 😄 .